app/src/main/java/com/framgia/music_50/data/source/remote/GetTracksAsyncTask.java

```diff
@@ -0,0 +1,80 @@
+package com.framgia.music_50.data.source.remote;
+
+import android.os.AsyncTask;
+import com.framgia.music_50.data.model.Track;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+import static com.framgia.music_50.utils.Constant.GET_METHOD;
+import static com.framgia.music_50.utils.Constant.REQUEST_TIMEOUT;
+
+public class GetTracksAsyncTask extends AsyncTask<String, Void, List<Track>> {
+    private OnFetchDataListener<Track> mListener;
+
+    public GetTracksAsyncTask(OnFetchDataListener<Track> listener) {
+        this.mListener = listener;
+    }
+
+    @Override
+    protected List<Track> doInBackground(String... strings) {
+        List<Track> tracks = null;
+        try {
+            String json = getDataFromUrl(strings[0]);
+            tracks = readDataFromJson(json);
+        } catch (IOException | JSONException e) {
+            mListener.onError(e);
+        }
+        return tracks;
+    }
+
+    @Override
+    protected void onPostExecute(List<Track> data) {
+        super.onPostExecute(data);
+        mListener.onSuccess(data);
+    }
+
+    private String getDataFromUrl(String s) throws IOException {
+        URL url = new URL(s);
+        HttpURLConnection connection = (HttpURLConnection) url.openConnection();
+        connection.setRequestMethod(GET_METHOD);
+        connection.setReadTimeout(REQUEST_TIMEOUT);
+        connection.setConnectTimeout(REQUEST_TIMEOUT);
+        connection.connect();
+        BufferedReader bufferedReader =
+                new BufferedReader(new InputStreamReader(connection.getInputStream()));
+        StringBuilder builder = new StringBuilder();
+        String line;
+        while ((line = bufferedReader.readLine()) != null) {
+            builder.append(line);
+        }
+        return builder.toString();
+    }
+
+    private List<Track> readDataFromJson(String json) throws JSONException {
+        List<Track> tracks = new ArrayList<>();
+        JSONObject rootObject = new JSONObject(json);
+        JSONArray jsonArray = rootObject.getJSONArray(Track.TrackEntry.COLLECTION);
+        for (int i = 0; i < jsonArray.length(); i++) {
+            JSONObject object = jsonArray.getJSONObject(i).getJSONObject(Track.TrackEntry.TRACK);
+            Track.TrackBuilder builder = new Track.TrackBuilder();
+            builder.setTitle(object.getString(Track.TrackEntry.TITLE));
```

Comment:
use like this :"Computer comp = new Computer.ComputerBuilder("500 GB", "2 GB").setBluetoothEnabled(true)
        .setGraphicsCardEnabled(true).build();"